File: src/sql/FSqlData.ts
import { FDecimal } from "../primitive/FDecimal";

export interface FSqlData {
	readonly asBoolean: boolean;
	readonly asBooleanNullable: boolean | null;

	readonly asString: string;
	readonly asStringNullable: string | null;

	readonly asInteger: number;
	readonly asIntegerNullable: number | null;

	readonly asNumber: number;
	readonly asNumberNullable: number | null;

	readonly asDecimal: FDecimal;
	readonly asDecimalNullable: FDecimal | null;

	readonly asDate: Date;
	readonly asDateNullable: Date | null;

	readonly asBinary: Uint8Array;
	readonly asBinaryNullable: Uint8Array | null;

	readonly asObject: any;
	readonly asObjectNullable: any | null;
}

File: lib/sql/f_sql_data.dart
// Copyright 2023, the Freemework.ORG project authors. Please see the AUTHORS
// file for details. All rights reserved. Use of this source code is governed
// by a BSD-style license that can be found in the LICENSE file.

import "dart:typed_data" show Uint8List;

import "../primitive/f_decimal.dart" show FDecimal;

abstract class FSqlData {
  Uint8List get asBinary;
  Uint8List? get asBinaryNullable;

  bool get asBoolean;
  bool? get asBooleanNullable;

  String get asString;
  String? get asStringNullable;

  int get asInteger;
  int? get asIntegerNullable;

  DateTime get asDate;
  DateTime? get asDateNullable;

  FDecimal get asDecimal;
  FDecimal? get asDecimalNullable;

  const FSqlData._(); // disallow extends (allow implements only)
}
